lib/plugins/rotate_asg_instances/asg.rb

```diff
@@ -48,6 +64,35 @@ def outdated_instances
 
     private
 
+    def execute_ssh(ssh_debug: false)
+      Retriable.retriable(base_interval: 5, tries: 3) do
+        @command_response = run_ssh_command(
+          first_ec2_instance, '/bin/true', ssh_debug: ssh_debug
+        )
+        raise StandardError.new, 'Validation failed.' unless
+          @command_response.exitstatus.zero?
+        @ilog.info(@command_response.output.split(/\r\n/)) if ssh_debug
+        @command_response.exitstatus.zero?
+      end
+    end
+
+    def run_ssh_command(instance, command, timeout_connection: true,
+                        ssh_debug: false)
+
+      remote_cmd =
+        build_command(instance, command, timeout_connection, ssh_debug)
+      Moonshot::SSHForkExecutor.new.run(remote_cmd)
+    end
+
+    def build_command(instance, command, timeout_connection, ssh_debug)
```

Comment:
There's a SSHCommandBuilder class `lib/moonshot/ssh_command_builder.rb` which does things that some methods of this class is doing

Reply:
@marciotoshio the `command_builder` does not support `options` or `timeout_connection` to pass and whether to use verbose which the pre-check needs. It is better if we maintain our original implementation here.